internal/drivers/nomad/driver.go

```diff
@@ -335,82 +351,89 @@ func (p *config) resourceJob(cpus, memGB int, vm, accountID string) (job *api.Jo
+	liteEngineHostPort = alloc.Resources.Networks[0].DynamicPorts[0].Value
+	for i := 1; i < len(alloc.Resources.Networks[0].DynamicPorts); i++ {
+		ports = append(ports, alloc.Resources.Networks[0].DynamicPorts[i].Value)
+	}
 
 	// sanity check
-	if port <= 0 || port > 65535 {
-		err = fmt.Errorf("scheduler: port %d generated is not a valid port", port)
+	if liteEngineHostPort <= 0 || liteEngineHostPort > 65535 {
+		err = fmt.Errorf("scheduler: lite engine host port %d generated is not a valid port", liteEngineHostPort)
 		logr.Errorln(err)
-		return ip, nodeID, port, err
+		return ip, nodeID, liteEngineHostPort, ports, err
+	}
+	for _, port := range ports {
+		if port <= 0 || port > 65535 {
+			err = fmt.Errorf("scheduler: gitspace host port %d generated is not a valid port", port)
+			logr.Errorln(err)
+			return ip, nodeID, liteEngineHostPort, ports, err
+		}
 	}
 
 	n, _, err := p.client.Nodes().Info(nodeID, &api.QueryOptions{})
 	if err != nil {
 		logr.WithError(err).Errorln("scheduler: could not get information about the node which picked up the resource job")
-		return ip, nodeID, port, err
+		return ip, nodeID, liteEngineHostPort, ports, err
 	}
 
 	ip = strings.Split(n.HTTPAddr, ":")[0]
 	if net.ParseIP(ip) == nil {
 		err = fmt.Errorf("scheduler: could not parse client machine IP: %s", ip)
 		logr.Errorln(err)
-		return ip, nodeID, port, err
+		return ip, nodeID, liteEngineHostPort, ports, err
 	}
 
-	return ip, nodeID, port, nil
+	return ip, nodeID, liteEngineHostPort, ports, nil
 }
 
 // initJob creates a job which is targeted to a specific node. The job does the following:
 //  1. Starts a VM with the provided config
 //  2. Runs a startup script inside the VM
-func (p *config) initJob(vm, startupScript string, hostPort int, nodeID string, resource cf.NomadResource) (job *api.Job, id, group string) {
+func (p *config) initJob(vm, startupScript string, liteEngineHostPort int, gitspacesPortMappings map[int]int, nodeID string, resource cf.NomadResource) (job *api.Job, id, group string) {
 	id = initJobID(vm)
 	group = fmt.Sprintf("init_task_group_%s", vm)
 	encodedStartupScript := base64.StdEncoding.EncodeToString([]byte(startupScript))
 
 	hostPath := fmt.Sprintf("/usr/local/bin/%s.sh", vm)
 	vmPath := fmt.Sprintf("/usr/bin/%s.sh", vm)
 
-	runCmd := fmt.Sprintf("%s run %s --name %s --cpus %s --memory %sGB --size %s --ssh --runtime=docker --ports %d:%s --copy-files %s:%s",
-		ignitePath,
-		p.vmImage,
-		vm,
-		resource.Cpus,
-		resource.MemoryGB,
-		resource.DiskSize,
-		hostPort,
-		strconv.Itoa(lehelper.LiteEnginePort),
-		hostPath,
-		vmPath)
+	var runCmdFormat string
+	runCmdFormat = "%s run %s --name %s --cpus %s --memory %sGB --size %s --ssh --runtime=docker --ports %d:%s --copy-files %s:%s"
+	args := []interface{}{ignitePath, p.vmImage, vm, resource.Cpus, resource.MemoryGB, resource.DiskSize, liteEngineHostPort, strconv.Itoa(lehelper.LiteEnginePort), hostPath, vmPath}
+	for vmPort, hostPort := range gitspacesPortMappings {
```

Comment:
how are we making sure that it will be called only for CDE ?

Reply:
For non-gitspace, gitspacesPortMappings is an empty map, execution will not go into the for loop